packages/fxa-payments-server/src/routes/Product/index.tsx

```diff
@@ -237,7 +239,47 @@ export const Product = ({
       productId
     );
 
+    const webSubscriptions = (customerSubscriptions || []).filter((s) =>
+      isWebSubscription(s)
+    ) as WebSubscription[];
+
+    const alreadySubscribedToSelectedPlan = customerIsSubscribedToPlan(
+      webSubscriptions,
+      selectedPlan
+    );
+
+    const planUpdateEligibilityResult = subscriptionUpdateEligibilityResult(
```

Comment:
As discussed in chat, should this be replaced with the new plan-eligibility API?

Reply:
Discussed with @chelsea-lewis. This will be implemented in a follow up ticket. I'll comment more in Jira with details.